tests/test_flow/test_persistence.py

```diff
@@ -821,3 +821,58 @@ def y():
 
         flow = builder.build()
         assert flow.get('y') == 1
+
+
+def test_volatile_ancestor(builder):
+    builder.assign('x', 5)
+
+    @builder
+    @bn.volatile(True)
+    @count_calls
+    def x_plus_one(x):
+        return x + 1
+
+    @builder
+    @bn.persist(False)
+    @count_calls
+    def x_plus_two(x_plus_one):
+        return x_plus_one + 1
+
+    @builder
+    @count_calls
+    def x_plus_three(x_plus_two):
+        return x_plus_two + 1
+
+    @builder
+    @count_calls
+    def x_plus_four(x_plus_three):
+        return x_plus_three + 1
+
+    flow = builder.build()
+    assert flow.get('x_plus_four') == 9
+    assert x_plus_one.times_called() == 1
+    assert x_plus_two.times_called() == 1
+    assert x_plus_three.times_called() == 1
+    assert x_plus_four.times_called() == 1
+    # In the same flow, volatile is not recomputed.
+    assert flow.get('x_plus_four') == 9
```

Comment:
Maybe we should test accessing the volatile entity directly as well? (`flow.get('x_plus_one')`)

Reply:
Sure, I'll add it